### PR TITLE
Feature/bazooka/referrals

### PR DESCRIPTION
**Pull request by julia-andersson**
## [APP-XXX]

- Move Forever/referrals from Giraffe to Octopus
- New schema

## Checklist

- [ ] Dark/light mode verification
- [ ] Landscape verification
- [ ] iPad verification
- [ ] iPod/iPhone 12 mini/iPhone SE verification

### CODE REVIEW

File: Projects/Forever/GraphQL/Octopus/MemberReferralInformationCodeUpdate.graphql [hunk 1]
```diff
@@ -0,0 +1,26 @@
+mutation MemberReferralInformationCodeUpdate($code: String!) {
+  memberReferralInformationCodeUpdate(code: $code) {
+    referralInformation {
```
Comment (sladan-hedvig) [inline Projects/Forever/GraphQL/Octopus/MemberReferralInformationCodeUpdate.graphql:3]: we don't use this referralInformation in code. so you can remove it.
We only use userError

File: Projects/Forever/GraphQL/Octopus/MemberReferralnformation.graphql [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+                    ...MonetaryAmountFragment
+                }
+                referrals {
+                    ...MemberReferralFragment
+                }
+        }
+    }
+}
+
+fragment MonetaryAmountFragment on Money {
```
Comment (sladan-hedvig) [inline Projects/Forever/GraphQL/Octopus/MemberReferralnformation.graphql:29]: we already have MoneyFragment

File: Projects/Forever/Sources/Components/InivtationTable.swift [hunk 3]
```diff
@@ -57,20 +57,21 @@ struct InvitationTable: View {
...
-                        ForEach(foreverData.invitations, id: \.hashValue) { row in
+                        ForEach(foreverData.referrals, id: \.hashValue) { row in
                             InvitationRow(row: row)
                         }
-                        getOtherDiscountsRow(foreverData)
-                        getTotalRow(foreverData)
                     }
-                    .sectionContainerStyle(.transparent)
-                    .padding(.horizontal, -16)
+                    getOtherDiscountsRow(foreverData)
```
Comment (sladan-hedvig) [inline Projects/Forever/Sources/Components/InivtationTable.swift:69]: please check this function getOtherDiscountsRow()
 if let otherDiscounts = foreverData.otherDiscounts { <- it looks odd if we have -0 for "your other discounts"

}

File: Projects/Forever/Sources/Components/InivtationTable.swift [hunk 4]
```diff
@@ -127,18 +128,17 @@ struct InvitationTable: View {
...
         hRow {
             HStack(spacing: 8) {
                 Circle().fill(row.statusColor).frame(width: 14, height: 14)
                 VStack(alignment: .leading) {
                     hText(row.name).foregroundColor(hTextColorNew.primary)
-                    if row.invitedByOther {
-                        hText(L10n.ReferallsInviteeStates.invitedYou, style: .subheadline)
-                            .foregroundColor(row.invitedByOtherLabelColor)
-                    }
+                    // TODO
```
Comment (sladan-hedvig) [inline Projects/Forever/Sources/Components/InivtationTable.swift:139]: What we have todo here?

File: Projects/Forever/Sources/Components/InivtationTable.swift [hunk 5]
```diff
@@ -57,20 +57,21 @@ struct InvitationTable: View {
...
-                        getTotalRow(foreverData)
                     }
-                    .sectionContainerStyle(.transparent)
-                    .padding(.horizontal, -16)
+                    getOtherDiscountsRow(foreverData)
+                    getTotalRow(foreverData)
                 }
+                .sectionContainerStyle(.transparent)
+                .padding(.horizontal, -16)
+                //                }
```
Comment (sladan-hedvig) [inline Projects/Forever/Sources/Components/InivtationTable.swift:74]: Remove comment

File: Projects/Forever/Sources/Store/ForeverStore.swift [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+    ) -> FiniteSignal<ForeverAction>? {
+        switch action {
+        case .fetch:
+            return FiniteSignal { callback in
+                let disposeBag = DisposeBag()
+                let query = OctopusGraphQL.MemberReferralInformationQuery()
+                disposeBag += self.octopus.client.fetch(query: query, cachePolicy: .fetchIgnoringCacheCompletely)
+                    .onValue { data in
+                        let grossAmount = data.currentMember.insuranceCost.monthlyGross
+                        let grossAmountMonetary = MonetaryAmount(
```
Comment (sladan-hedvig) [inline Projects/Forever/Sources/Store/ForeverStore.swift:24]: When you switch to MoneyFragment in query you can just init with
 MonetaryAmount(fragment: grossAmount.fragments.moneyFragment) etc

**Review by sladan-hedvig — CHANGES_REQUESTED** [review]

File: Projects/Forever/Sources/Components/InivtationTable.swift (hunk 3 above)
Comment (julia-andersson) [reply Projects/Forever/Sources/Components/InivtationTable.swift:70]: Agree. Will change this.

**Review by sladan-hedvig — APPROVED** [review]

**Pull request merged**